scylla/src/transport/errors.rs

```diff
@@ -225,6 +234,16 @@ pub enum NewSessionError {
     RequestTimeout(String),
 }
 
+/// A protocol error.
+///
+/// It indicates an inconsistency between CQL protocol
+/// and server's behavior.
+/// In some cases, it could also represent a misuse
+/// of internal driver API - a driver bug.
```

Comment:
Well, in some cases it's really hard to distinguish whose fault it is - look at the `NonfinishedPagingState` for example. It suggests that this MAY be a driver's internal API misuse.